dash/resources.py

```diff
@@ -2,20 +2,53 @@
 import warnings
 import os
 
+import typing as _t
+import typing_extensions as _tx
+
+
 from .development.base_component import ComponentRegistry
 from . import exceptions
 
 
+# ResourceType has `async` key, use the init form to be able to provide it.
+ResourceType = _tx.TypedDict(
+    "ResourceType",
+    {
+        "namespace": str,
+        "async": _t.Union[bool, _t.Literal["eager", "lazy"]],
```

Comment:
so legal values are `True`, `False`, `"eager"`, and `"lazy"`? Are these documented somewhere?

Reply:
Yes those are the valid values, along with None/undefined for all of them properties since `total=False` set them as optional.
This is mostly for usage of `async` chunks of components libraries (dcc.Graph, dcc.Markdown, dash_table.DataTable). The usual value is `True` or undefined, eager or lazy are abstractions that are not very well documented, it is commented here:
https://github.com/plotly/dash/blob/d26c9d37c4b50b001588ff6b2862baf0c66dbe85/dash/resources.py#L66-L72

With `Dash(eager_loading=True)`, the `async: "eager"` chunks normally loaded when the component mounts, would be loaded on page load.